timelock-impl/src/test/java/com/palantir/atlasdb/timelock/transaction/client/DistributingModulusGeneratorTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.transaction.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+import com.google.common.collect.Iterables;
+
+public class DistributingModulusGeneratorTest {
+    private DistributingModulusGenerator generator;
+
+    @Test
+    public void doesNotReuseResiduesIfNotNeeded() {
+        setupGeneratorWithModulus(3);
+
+        requestResiduesAndAssertBalanced(3);
```

Comment:
I would argue being slightly more verbose in your tests. Reading this, and looking at the test name, I don't really have an idea what it is doing.